File: solutions/1528. Shuffle String/1528.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
 public:
  string restoreString(string_view s, const vector<int>& indices) {
    string ans(s.length(), '.');

    for (int i = 0; i < indices.size(); ++i)
      ans[indices[i]] = s[i];

    return ans;
  }
};

File: solutions/1528. Shuffle String/1528.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
  def restoreString(self, s: str, indices: List[int]) -> str:
    ans = [''] * len(s)

    for i, index in enumerate(indices):
      ans[index] = s[i]

    return ''.join(ans)